### PR TITLE
Fix printing of fun_expr as only argument of a function application.

### PR DESCRIPTION
**Pull request by IwanKaramazow**
Fixes https://github.com/facebook/reason/issues/1743

### CODE REVIEW

**Comment by chenglou** [comment]
Should we just go straight to the dot uncurried form instead?

**Comment by IwanKaramazow** [comment]
Can't sleep at night with a bug in master while I'm taking care of the dot uncurried form :p

File: src/reason-parser/reason_pprint_ast.ml [hunk 1]
```diff
@@ -6741,6 +6741,10 @@ class printer  ()= object(self:'self)
     let categorizeFunApplArgs args =
       let reverseArgs = List.rev args in
       match reverseArgs with
+      | (_, {pexp_desc = Pexp_fun _})::[] ->
+          (* If there's only one arg, should be printed as normal *)
+          (* e.g. Js.Option.andThen([@bs] (w => w#getThing())); *)
+          `NormalFunAppl args
       | ((_, {pexp_desc = Pexp_fun _}) as callback)::args
```
[the remark this answers is not in the document]
Comment (jordwalke) [reply src/reason-parser/reason_pprint_ast.ml]: I didn't notice the `when` below! Sorry for the noise.

**Comment by chenglou** [comment]
So is this good to go? Love the attitude btw

**Comment by IwanKaramazow** [comment]
Still need to add extra test cases, will be done in 24 hours.

**Comment by IwanKaramazow** [comment]
Done, added extra tests with attributes sitting on callbacks.

Examples:
```
describe([@attr] "App", [@attr] () =>
  test([@attr] "math", [@attr] () =>
    Expect.expect(1 + 2) |> toBe(3)
  )
);

Thing.map(
  foo,
  bar,
  baz,
  foo2,
  [@attr] bakjlksjdf,
  okokokok,
  [@attr] (abc, z) => {
    let x = 1;
    MyModuleBlah.toList(x, argument);
  }
);
```
Ready on my end

**Comment by jordwalke** [comment]
Thank you, as always!

**Pull request merged**